### PR TITLE
Move util/metautils to root-level package metadata, fixes #392

### LINKED ISSUE #392: v2: Move `metautils` to it's own package
This is the package that allows to pass things across gRPC peers. You mention @johanbrandhorst some other methods for this 🤔 Is there something or should we maintain it here?

### PR DESCRIPTION
**Pull request by rahulkhairwar**
- Move util/metautils to root-level package metadata
- Rename NiceMD to MD, which is a wrapper for grpc/metadata.MD

Fixes #392

### CODE REVIEW

**Comment by johanbrandhorst** [comment]
Hi Rahul, thanks for your PR. Have you signed the CLA at https://cla.developers.google.com/clas?

**Comment by rahulkhairwar** [comment]
hi @johanbrandhorst , sorry I missed it earlier. I've signed it now.

File: testing/testpb/v1/test.proto [hunk 1]
```diff
@@ -2,7 +2,7 @@ syntax = "proto3";
 
 package testing.testpb.v1;
 
-option go_package = ".;testpb";
+option go_package = "./;testpb";
```
Comment (johanbrandhorst) [inline testing/testpb/v1/test.proto:5]: Why was this change made?

**Review by johanbrandhorst — CHANGES_REQUESTED** [review]
Hi, the CLA bot still doesn't seem to have recognized your signature, could you please share a screenshot of the signed CLA from the website? Thanks

File: testing/testpb/v1/test.proto (hunk 1 above)
Comment (rahulkhairwar) [reply testing/testpb/v1/test.proto:5]: I couldn't run `make all` without this change, see below:
Screenshot 2021-12-05 at 11 57 35 PM

I guess different versions of `protoc-gen`.
Let me revert this change and push again.

**Comment by rahulkhairwar** [comment]
Also, for the CLA, please look at the screenshot below:
Screenshot 2021-12-06 at 12 00 50 AM

File: interceptors/auth/auth_test.go [hunk 2]
```diff
@@ -9,19 +9,20 @@ import (
 	"testing"
 	"time"
 
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/metadata"
```
Comment (johanbrandhorst) [inline interceptors/auth/auth_test.go:12]: This needs to be moved into the 3rd import group with the other imports.

File: interceptors/auth/metadata.go [hunk 3]
```diff
@@ -7,10 +7,10 @@ import (
 	"context"
 	"strings"
 
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/metadata"
```
Comment (johanbrandhorst) [inline interceptors/auth/metadata.go:10]: I can see you moved all of these imports around, I think we want to move them down again.

File: interceptors/auth/metadata_test.go [hunk 4]
```diff
@@ -7,62 +7,62 @@ import (
 	"context"
 	"testing"
 
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/metadata"
```
Comment (johanbrandhorst) [inline interceptors/auth/metadata_test.go:10]: Move this down to the bottom again please

File: interceptors/retry/retry.go [hunk 5]
```diff
@@ -10,13 +10,13 @@ import (
 	"sync"
 	"time"
 
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/metadata"
```
Comment (johanbrandhorst) [inline interceptors/retry/retry.go:13]: Move this down to the bottom again please

File: interceptors/skip/interceptor_test.go [hunk 6]
```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/stretchr/testify/suite"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/metadata"
+	grpcMetadata "google.golang.org/grpc/metadata"
```
Comment (johanbrandhorst) [inline interceptors/skip/interceptor_test.go:17]: Do we really need to alias this import here? It's not ambiguous in this file.

File: testing/testpb/test_grpc.pb.go [hunk 7]
```diff
@@ -4,7 +4,6 @@ package testpb
 
 import (
 	context "context"
-
```
Comment (johanbrandhorst) [inline testing/testpb/test_grpc.pb.go:7]: Can you undo these generated file changes please? I don't expect them to change in this PR.

File: testing/testpb/test.pb.go [hunk 8]
```diff
@@ -1,18 +1,16 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
-// 	protoc-gen-go v1.25.0
+// 	protoc-gen-go v1.25.0-devel
```
Comment (johanbrandhorst) [inline testing/testpb/test.pb.go:3]: Please undo these generated file changes.

**Review by johanbrandhorst — CHANGES_REQUESTED** [review]
Thanks for signing the CLA. There are still some outstanding issues here.

File: interceptors/auth/auth_test.go (hunk 2 above)
Comment (rahulkhairwar) [reply interceptors/auth/auth_test.go]: Fixed.

File: interceptors/auth/metadata.go (hunk 3 above)
Comment (rahulkhairwar) [reply interceptors/auth/metadata.go]: Fixed.

File: interceptors/auth/metadata_test.go (hunk 4 above)
Comment (rahulkhairwar) [reply interceptors/auth/metadata_test.go]: Fixed.

File: interceptors/retry/retry.go (hunk 5 above)
Comment (rahulkhairwar) [reply interceptors/retry/retry.go]: Fixed.

File: testing/testpb/test_grpc.pb.go (hunk 7 above)
Comment (rahulkhairwar) [reply testing/testpb/test_grpc.pb.go]: Reverted these changes.

File: testing/testpb/test.pb.go (hunk 8 above)
Comment (rahulkhairwar) [reply testing/testpb/test.pb.go]: Reverted these changes.

File: interceptors/skip/interceptor_test.go (hunk 6 above)
Comment (johanbrandhorst) [reply interceptors/skip/interceptor_test.go:17]: It's probably fine, lets leave it

**Review by johanbrandhorst — APPROVED** [review]
LGTM!

**Comment by johanbrandhorst** [comment]
Thank you for your contribution!

**Pull request merged**